File: Racharla Akhil - PB002/Wiki-search-node/WikiSearch.js
start()

function start() {
	var prompt = require('prompt')
	if (process.argv.length <= 2) {
		prompt.get(['keyword', 'fileName'], function (err, result) {
	     	var keyword = result.keyword
	     	var fileName = result.fileName
	     	saveURL(keyword, fileName)
	 	})
	} else if (process.argv.length == 3) {
		prompt.get(['keyword'], function (err, result) {
	     	var keyword = result.keyword
	     	saveURL(keyword, process.argv[2])
	 	})
	} else {
			saveURL(process.argv[3], process.argv[2])
	}
}

function saveURL(keyword, fileName) {
	if (keyword.length > 0 && fileName.length > 0) {
		createAndSendResponse(keyword, fileName)
	}
}

function createAndSendResponse(keyword, fileName) {
	var request = require('request')
	var fs = require('fs')
	request({
		url: 'https://en.wikipedia.org/w/api.php?action=opensearch&search=' + keyword,
		json: true
	}, function (error, response, body) {
		if (!error && response.statusCode == 200 && response.body.length >= 4 && response.body[3].length > 0) {
		   	console.log(keyword + ": " + response.body[3][0])
		   	fs.appendFile(fileName + '.log',  keyword + ": " + response.body[3][0] + '\n', function (err) {})
		} else {
			console.log(keyword + ": No Result")
		   	fs.appendFile(fileName + '.log',  keyword + ": No Result" )
		}
	})
}

File: Racharla Akhil - PB002/Wiki-search.py
import argparse
import json
import requests

def main():
	parser = argparse.ArgumentParser(description='Search result for top title in wiki search bar')

	parser.add_argument('-f', '--fileName', type=str, help='logfile name in which result will be saved')
	parser.add_argument('-k', '--keyword', type=str, help='keyword to be searched')
	args = parser.parse_args()

	#search keyword
	keyword = ''
	if args.keyword != '':
		keyword = args.keyword
		
	if keyword is None:
		keyword = raw_input('Enter keyword to search: ')

	#fileName name
	fileName = ''
	if args.fileName != '':
		fileName = args.fileName

	if fileName is None:
		fileName = raw_input('Enter name of log file: ')

	file = open(fileName + '.log','a')
	res = requests.get('https://en.wikipedia.org/w/api.php?action=opensearch&search=' + keyword)
	jsonResults = json.loads(str(res.text))
	if len(jsonResults) >= 4 and len(jsonResults[3]) >= 1:
		print keyword + ': ' + jsonResults[3][0]
		file.write(keyword + ': ' + jsonResults[3][0])
		file.write('\n')
	else :
		print keyword + ': No Result'
		file.write(keyword + ': No Result')
		file.write('\n')

if __name__ == '__main__':
	main()